app/controllers/posts_controller.rb

```diff
@@ -25,11 +25,12 @@ def edit; end
   # POST /posts or /posts.json
   def create
     @post = Post.new(post_params)
-
+    @user = current_user
+    @post.author_id = @user.id
     respond_to do |format|
       if @post.save
-        format.html { redirect_to post_url(@post), notice: 'Post was successfully created.' }
-        format.json { render :show, status: :created, location: @post }
+        format.html { redirect_to user_post_path(@user, @post), notice: 'Post was successfully created.' }
+        format.json { render :index, status: :created, location: @post }
       else
         format.html { render :new, status: :unprocessable_entity }
         format.json { render json: @post.errors, status: :unprocessable_entity }
```

Comment:
- [ ] Please your post form is not rendering as you can see below, so make sure that you fix that and ensure that creating a post, commenting on a post, and liking a post works perfectly:+1: 
  
  [Screencast from 07-21-2023 05:57:24 PM.webm](https://github.com/MarwanDev/blog-app/assets/92303891/588bcc33-3503-4e83-91d8-98d6594f213f)